tests/botserver_mock.py

```diff
@@ -0,0 +1,170 @@
+import json
+import threading
+import time
+import websocket
+
+import handlers.handler_factory as handler_factory
+from handlers import *
+from util.loghandler import *
+from tests.slackwrapper_mock import SlackWrapperMock
+from bottypes.invalid_console_command import *
+from slackclient.server import SlackConnectionError
+
+
+class BotServerMock():
+    user_list = {}
+
+    def __init__(self):
+        log.debug("Parse config file and initialize threading...")
+        self.running = False
+        self.config = {}
+        self.bot_name = "unittest_bot"
+        self.bot_id = "unittest_botid"
+        self.bot_at = "@unittest_bot"
+        self.slack_wrapper = SlackWrapperMock("unittest_apikey")
+
+    def quit(self):
+        """Inform the application that it is quitting."""
+        self.running = False
+
+    def load_config(self):
+        """Load configuration file."""
+        self.config = {
+            "bot_name": "unittest_bot",
+            "api_key": "unittest_apikey",
+            "send_help_as_dm": "1",
+            "admin_users": [
+                "admin_user"
+            ],
+            "auto_invite": [],
+            "wolfram_app_id": "wolfram_dummyapi"
+        }
+
+    def get_config_option(self, option):
+        """Get configuration option."""
+        result = self.config.get(option)
+
+        return result
+
+    def set_config_option(self, option, value):
+        """Set configuration option."""
+
+        if option in self.config:
+            self.config[option] = value
+            log.info("Updated configuration: {} => {}".format(option, value))
+        else:
+            raise InvalidConsoleCommand("The specified configuration option doesn't exist: {}".format(option))
+
+    def parse_slack_message(self, message_list):
+        """
+        The Slack Real Time Messaging API is an events firehose.
+        Return (message, channel, ts, user) if the message is directed at the bot,
+        otherwise return (None, None, None, None).
+        """
+        for msg in message_list:
+            if msg.get("type") == "message" and "subtype" not in msg:
+                if self.bot_at in msg.get("text", ""):
+                    # Return text after the @ mention, whitespace removed
```

Comment:
repeated code. Note there's a PR to also get rid of `!` after the bot name. Could we move this parser into a generic function so that we don't have to work to keep the Mock and the real bot in sync? Maybe we can make the real bot mockable by instead mocking the slack wrapper, so that the parsing and stuff is always matching?